hoomd/hpmc/IntegratorHPMCMonoGPU.h

```diff
@@ -186,49 +232,56 @@ class IntegratorHPMCMonoGPU : public IntegratorHPMCMono<Shape>
         GlobalArray<unsigned int> m_reject_out;               //!< Flags to reject particle moves, per particle (temporary)
 
-        GlobalArray<unsigned int> m_nlist;                       //!< List of overlapping particles
-        GlobalArray<unsigned int> m_nneigh;                     //!< Number of neighbors
+        GlobalArray<unsigned int> m_n_depletants;             //!< List of number of depletants, per particle
+        GlobalArray<unsigned int> m_n_depletants_ntrial;      //!< List of number of depletants, per particle, trial insertion and configuration:w
+        GlobalArray<int> m_deltaF_int;                        //!< Free energy difference delta_F per particle for MH, rescaled to int
+        unsigned int m_max_len;                               //!< Max length of shared memory allocation per group
+        GlobalArray<unsigned int> m_req_len;                  //!< Requested length of shared mem per group
+
         detail::UpdateOrderGPU m_update_order;                   //!< Particle update order
-        unsigned int m_maxn;                                     //!< Max number of neighbors
-        GlobalArray<unsigned int> m_overflow;                    //!< Overflow condition for neighbor list
-        GlobalArray<unsigned int> m_condition;                  //!< Condition of acceptance kernel
+        GlobalArray<unsigned int> m_condition;                  //!< Condition of convergence check
 
         //! For energy evaluation
-        unsigned int m_maxn_patch;                            //!< Maximum number of patch neighbors
-        GlobalArray<unsigned int> m_overflow_patch;           //!< Overflow condition for neighbor list
-        GlobalArray<unsigned int> m_nlist_patch_old;          //!< List of neighbors in old config
-        GlobalArray<float> m_energy_old;                      //!< Energy contribution per neighbor in old config
-        GlobalArray<unsigned int> m_nneigh_patch_old;         //!< Number of neighbors in old config
-        GlobalArray<unsigned int> m_nlist_patch_new;          //!< List of neighbors in new config
-        GlobalArray<float> m_energy_new;                      //!< Energy contribution per neighbor in new config
-        GlobalArray<unsigned int> m_nneigh_patch_new;         //!< Number of neighbors in new config
         GlobalArray<Scalar> m_additive_cutoff;                //!< Per-type additive cutoffs from patch potential
 
         GlobalArray<hpmc_counters_t> m_counters;                    //!< Per-device counters
         GlobalArray<hpmc_implicit_counters_t> m_implicit_counters;  //!< Per-device counters for depletants
 
+        std::vector<hipStream_t> m_narrow_phase_streams;             //!< Stream for narrow phase kernel, per device
+        std::vector<std::vector<hipStream_t> > m_depletant_streams;  //!< Stream for every particle type, and device
+        std::vector<std::vector<hipStream_t> > m_depletant_streams_phase1;  //!< Streams for phase1 kernel
```

Comment:
Could you document somewhere that the 2-phase process is specifically for when ntrial > 0? There's a lot of code here and it's hard to keep track of, and will be more so for anyone who's unfamiliar with the sampling process to begin with.

Reply:
I've added a comment to that extent

https://github.com/glotzerlab/hoomd-blue/blob/1d0c644f74b3922dad8dd59a64305d6266142c97/hoomd/hpmc/IntegratorHPMCMonoGPU.h#L268

However, ntrial = 0 will (likely) be eliminated in subsequent PRs to avoid having two depletants code paths